Album/ViewControllers/ProfileControllers/CloseFriendViewController.m

```diff
@@ -8,134 +8,146 @@
     // Fetch friends
-    PFQuery *friendQuery = [PFQuery queryWithClassName:classNameFriendship];
-    NSLog(@"%@",self.user.objectId);
-    [friendQuery whereKey:@"requesterId" equalTo:self.user.objectId];
-    [friendQuery whereKey:@"hasFriended" equalTo:@(2)];
-    [friendQuery findObjectsInBackgroundWithBlock:^(NSArray *friendships, NSError *error) {
-        self.friendshipArr = friendships;
-        NSMutableArray *friendArr = [[NSMutableArray alloc] init];
-        NSMutableArray *otherFriendships = [[NSMutableArray alloc] init];
-        if (friendships != nil) {
-            for (Friendship *friendship in friendships) {
-                NSString *friendId = friendship[@"recipientId"];
-                PFQuery *userQuery = [PFUser query];
-                [userQuery whereKey:@"objectId" equalTo:friendId];
-                PFUser *user = [userQuery findObjects][0];
-                [friendArr addObject:user];
-                
-                // Fetch friendship from other side
-                PFQuery *query = [PFQuery queryWithClassName:classNameFriendship];
-                [query whereKey:@"recipientId" equalTo:self.user.objectId];
-                [query whereKey:@"requesterId" equalTo:user.objectId];
-                [otherFriendships addObject:[query findObjects][0]];
-            }
-            self.friendsArr = friendArr;
-            self.otherFriendshipArr = otherFriendships;
-            dispatch_async(dispatch_get_main_queue(), ^{
-                            NSLog(@"AYYYY");
-                            [self.tableView reloadData];
-                self.tableView.rowHeight = UITableViewAutomaticDimension;
-                        });
-        } else {
-            NSLog(@"%@",
-                  error.localizedDescription);
+    NSArray *friendships = [ParseAPIHelper fetchFriendships:self.user.objectId];
+    self.friendshipArr = friendships;
+    self.filterFriendshipArr = friendships;
+    NSMutableArray *friendArr = [[NSMutableArray alloc] init];
+    NSMutableArray *otherFriendships = [[NSMutableArray alloc] init];
+    if (friendships != nil) {
+        for (Friendship *friendship in friendships) {
+            NSString *friendId = friendship[@"recipientId"];
+            PFQuery *userQuery = [PFUser query];
+            [userQuery whereKey:@"objectId" equalTo:friendId];
+            PFUser *user = [userQuery findObjects][0];
+            [friendArr addObject:user];
+
+            // Fetch friendship from other side
+            PFQuery *query = [PFQuery queryWithClassName:classNameFriendship];
+            [query whereKey:@"recipientId" equalTo:self.user.objectId];
+            [query whereKey:@"requesterId" equalTo:user.objectId];
+            [otherFriendships addObject:[query findObjects][0]];
         }
-    }];
+        self.friendsArr = friendArr;
+        self.filterFriendsArr = friendArr;
+        self.otherFriendshipArr = otherFriendships;
+        self.filterOtherFriendshipArr = otherFriendships;
```

Comment:
are this arrays the same?

Reply:
I was going to implement search bar but I think I will get rid of it